boards/nordic/nrf52840dk/nrf52840dk_nrf52840.dts

```diff
@@ -135,6 +135,16 @@
 	};
 };
 
+&dcdcen {
+	status = "okay";
+	regulator-boot-on;
+};
+
+&dcdcen0 {
+	status = "okay";
+	regulator-boot-on;
+};
```

Comment:
this doesn't seem to describe the regulator scheme properly

![image](https://github.com/zephyrproject-rtos/zephyr/assets/25011557/864c78eb-018e-46e4-883b-ce02f894eb18)


e.g., dcdcen0 is a register to enable REG0. I think this needs to be treated carefully and deserves some pre-study since it will have an impact on many boards.

cc: @aasinclair

Reply:
There is a voltage setting but that setting is part of the UICR not the power module, but yes support for having that set in dts should be added